### DISCUSSION TITLE
Undefined error when trying to read public key

### DISCUSSION THREAD

**Discussion by wmxvi** [openpgpjs/openpgpjs#1793] [Q&A] [answered]
Hi,

I'm trying to read a public key stored in my DB using the following code:

```
      this.partnerPublicKey = await openpgp.readKey({
        armoredKey: this.auth.publicKey,
      });
```

But the following error is thrown:
```

Cannot read properties of undefined (reading 'isPrototypeOf')
    at isStream (/app/shared/node_modules/openpgp/node_modules/@openpgp/web-stream-tools/lib/util.js:18:72)
    at stream.transformPair (/app/shared/node_modules/openpgp/node_modules/@openpgp/web-stream-tools/lib/streams.js:334:7)
    at /app/shared/node_modules/openpgp/src/encoding/armor.js:247:32
    at new Promise (<anonymous>)
    at unarmor (/app/shared/node_modules/openpgp/src/encoding/armor.js:235:10)
    at Object.exports.readKey (/app/shared/node_modules/openpgp/src/key/factory.js:312:34)
```

I have made sure that the key is in the following format:

```
-----BEGIN PGP PUBLIC KEY BLOCK-----

Key Contents

-----END PGP PUBLIC KEY BLOCK-----
```

Environment:

**Node:** v16.20.0
**Modules:**
  "openpgp": "5.11.2",
  "@types/openpgp": "5.11.2",
  "@openpgp/web-stream-tools": "^0.0.11-patch-0",

**Comment by twiss** [comment, marked as answer]
Hey :wave: It sounds like you're in a bit of a weird environment where `global.ReadableStream` exists but doesn't have a `prototype` property. Can that be?

You might need to remove/rename that global variable and/or do something like

```js
Object.assign(global, require('node:stream/web'));
```

(or use a newer version of Node.js.)

**Reply by wmxvi** [reply]
Wow super! Thank you that worked like a charm - Yes weird environment, will upgrade Node :)
